### DISCUSSION TITLE
how to use two equals methods for the same url in laravel routing?

### DISCUSSION THREAD

**Discussion by AdministratedAdmin** [community/community#76123] [Programming Help] [answered]
### Body

If a put a another put in the same url, only one is declared. How i can use the two same methods?

Code:

`Route::put('/acao/administrar', [AdmController::class, 'raprova'])->name('aprovar');
Route::put('/acao/administrar', [AdmController::class, 'rreprova'])->name('reprovar');
`

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by MegatIrfan** [comment]
In Laravel, when you define routes, the routes are matched in the order they are declared. In your case, both routes have the same URL (/acao/administrar) and HTTP method (PUT), so the second one will override the first one. This means that only the second route will be registered, and the first one will be ignored.

If you want to have two different actions for the same URL and HTTP method, you can consider combining them into a single controller method and handle the logic for both cases inside that method. For example:

Route::put('/acao/administrar', [AdmController::class, 'administrar'])->name('administrar');


And inside your AdmController, you can define the administrar method:
public function administrar()
{
    // Logic for approving
    // ...

    // Logic for rejecting
    // ...

    // You can use request data or any other conditions to determine the flow
}

**Reply by AdministratedAdmin** [reply]
How use this?

```
public function administrar() {
 
//aprovar 
 
$aprovar = //logic

//reprovar 

$reprovar = //logic

}
```


Like this? 

How i call the function in the form? Actually, i use `<form action="{{ route('reqaprova') -> $requerimento->id }}"></form>`

Sorry, im beginner in this

**Reply by MegatIrfan** [reply, marked as answer]
```
public function administrar(Request $request, $requerimentoId) {
    // Validate the request data, if needed
    $request->validate([
        // Add validation rules if necessary
    ]);

    // Get the action (approve or reject) from the form submission
    $action = $request->input('action');

    // Retrieve the requerimento based on the provided ID
    $requerimento = Requerimento::find($requerimentoId);

    if (!$requerimento) {
        // Handle the case where the requerimento is not found
        return redirect()->back()->with('error', 'Requerimento not found');
    }

    // Perform the appropriate action based on the form submission
    if ($action == 'approve') {
        // Logic for approval
        $aprovar = // your approval logic here
    } elseif ($action == 'reject') {
        // Logic for rejection
        $reprovar = // your rejection logic here
    } else {
        // Handle the case where an invalid action is submitted
        return redirect()->back()->with('error', 'Invalid action');
    }

    // Redirect back with a success message
    return redirect()->back()->with('success', 'Action successfully performed');
}

```
